File: clients/rust/src/send_message.rs
mod proto;

use proto::chat_package::Message;
use proto::chat_package::chat_service_client::ChatServiceClient;
use std::error::Error;
use tonic::Request;
use tonic::transport::{Channel, ClientTlsConfig};

const SERVER_ADDR: &str = "https://grpc-sample-35975833932.southamerica-west1.run.app"; // Demo server

#[tokio::main]
async fn main() -> Result<(), Box<dyn Error>> {
    let pid = std::process::id();
    // Connect to the gRPC server
    rustls::crypto::ring::default_provider().install_default().expect("Failed to install rustls crypto provider");
    let tls = ClientTlsConfig::new()
        .with_native_roots()
        .assume_http2(true);    
    let channel = Channel::from_static(SERVER_ADDR).tls_config(tls)?.connect().await?;
    let mut chat_service = ChatServiceClient::new(channel);

    // Build your request (replace fields as needed)
    let request = Request::new(Message {
        content: format!("Hi from PID {:?}. This is a test", pid).to_string(),
    });

    println!("Sending: {:?}", request);
    let response = chat_service.send_message(request).await?;
    println!("Resonse: {:?}", response);

    Ok(())
}

File: clients/go/send_message.go
package main

import (
	"context"
	"crypto/tls"
	"log"
	"time"

	proto "grpc-sample-client/protos"

	"fmt"
	"os"

	"google.golang.org/grpc"
	"google.golang.org/grpc/credentials"
)

func main() {
	var addr = "grpc-sample-35975833932.southamerica-west1.run.app" // Demo server
	var tlsConfig = &tls.Config{InsecureSkipVerify: true}
	// Set up a connection to the server.
	conn, err := grpc.NewClient(addr, grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig)))
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	defer conn.Close()
	gRPCService := proto.NewChatServiceClient(conn)

	// Contact the server and print out its response.
	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
	defer cancel()

	var request = proto.Message{Content: fmt.Sprintf("Hi from PID %d. This is a test", os.Getpid())}
	log.Printf("Sending: %v", &request)
	response, err := gRPCService.SendMessage(ctx, &request)
	if err != nil {
		log.Fatalf("Error: %v", err)
	}
	log.Printf("Response: %v", response)
}
